app/jobs/index_firm_job.rb

```diff
@@ -4,6 +4,10 @@ class IndexFirmJob < ActiveJob::Base
   end
 
   def perform(firm)
-    FirmRepository.new.store(firm) if firm.publishable?
+    if firm.publishable?
+      FirmRepository.new.store(firm)
+    else
+      DeleteFirmJob.perform_later(firm.id)
```

Comment:
My first thought here was- what happens if we try to delete something that doesn't exist in Elastic Search. Digging down into the ElasticSearchClient, it just returns if the delete HTTP request was ok or not. So looks good!